### DISCUSSION TITLE
Quasar uuid generated on component compilation breaks snapshots

### DISCUSSION THREAD

**Discussion by AxelMolina1** [quasarframework/quasar#16316] [General - Components / Directives / etc] [answered]
Hello !
I am currently implementing snapshot tests in my Quasar project and I had an issue.
When I am mounting a component with QFile or QBtnDropdown, I figured out that uuids are generated for aria-controls and id attributes of button and qfile, which breaks my test : 

![image](https://github.com/quasarframework/quasar/assets/86236852/a9042b78-9091-4f4b-9b2a-15e3b450fe28)

![image](https://github.com/quasarframework/quasar/assets/86236852/1d3e1b32-2602-41fd-8faf-08b0bcf301bb)

Does anyone have an idea of how to proceed ?
I tried to search a way to mock this process without success. I also tried to modify the html generated before testing in to remove those lines but I do not think that it's the proper way to do it. I am sure at least one person came across this problem before.

Thanks !

**Comment by pdanpdan** [comment, marked as answer]
Use `for` prop with a fixed id

**Reply by AxelMolina1** [reply]
I am not using `for` prop, it's auto generated by Quasar

**Reply by pdanpdan** [reply]
Use it

**Reply by AxelMolina1** [reply]
Okay thank you it worked! But I still have the issue with aria-controls of QBtnDropdown and I do not see it as a prop in Quasar doc. Do you have an idea?

**Comment by pdanpdan** [comment]
I have to search the source. I'm not sure we covered this :(

Yes, the only way would be to mock/replace `uid` function somehow.

**Reply by AxelMolina1** [reply]
Okay thank you for your help, I am going to try this!
